frontend/revive-gallery-app/src/components/products/ProductDetails.js

```diff
@@ -12,8 +12,32 @@ const ProductDetails = () => {
   const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
   const productId = searchParams.get('id');
+  const [product, setProduct] = useState({});
```

Comment:
set the default value to null and when the product is set then only show the product in the UI.